### PR TITLE
Update Workflows

### PR DESCRIPTION
**Pull request by pauladkisson**
- [x] remove conda
- [x] x64
- [x] arm64
- [x] python 3.12
- [x] dev tests

### CODE REVIEW

File: .github/workflows/run-tests.yml [hunk 1]
```diff
@@ -37,10 +37,9 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        python-version: [3.8, 3.9, "3.10", 3.11]
-        os: [ubuntu-latest, windows-latest, macos-latest]
+        python-version: [3.8, 3.9, "3.10", 3.11, 3.12]
```
Comment (CodyCBakerPhD) [inline .github/workflows/run-tests.yml:40]: It's petty, but what about having all these python versions in quotes so they look consistent?

**Comment by pauladkisson** [comment]
Looks like [scanimage-tiff-reader](https://pypi.org/project/scanimage-tiff-reader/) doesn't support arm...I guess we wait for them?

**Comment by CodyCBakerPhD** [comment]
> Looks like [scanimage-tiff-reader](https://pypi.org/project/scanimage-tiff-reader/) doesn't support arm...I guess we wait for them?

Can you ping them about that?

Also you can just add a skip for that on relevant tests, `platform.architecture` + `platform.platform` or `sys.platform` should give you something to use to distnguish skip logic

**Comment by pauladkisson** [comment]
> > Looks like [scanimage-tiff-reader](https://pypi.org/project/scanimage-tiff-reader/) doesn't support arm...I guess we wait for them?
> 
> Can you ping them about that?

Done: https://gitlab.com/vidriotech/scanimagetiffreader-python/-/issues/31

File: .github/workflows/run-tests.yml [hunk 2]
```diff
@@ -75,7 +74,16 @@ jobs:
...
           key: ophys-datasets-042023-${{ matrix.os }}-${{ steps.ophys.outputs.HASH_OPHYS_DATASET }}
 
-      - name: Run full pytest with coverage
+      - name: Run full pytest (macos-latest)
+        if: ${{ matrix.os == 'macos-latest' }}
+        run: |
+          pytest -n auto --dist loadscope \
+            --ignore=tests/test_scan_image_tiff.py \
+            --ignore=tests/test_scanimage_utils.py \
+            --ignore=tests/test_scanimagetiffimagingextractor.py \
```
Comment (CodyCBakerPhD) [inline .github/workflows/run-tests.yml:83]: Were you not able to get a skip to work well in the test suite itself? (like `pytest.mark.skipif`)?

File: tests/test_scan_image_tiff.py [hunk 3]
```diff
@@ -5,7 +5,12 @@
 
 from hdmf.testing import TestCase
 from numpy.testing import assert_array_equal
-from ScanImageTiffReader import ScanImageTiffReader
+import platform
+
+if (
+    platform.processor() != "arm"
```
Comment (CodyCBakerPhD) [inline tests/test_scan_image_tiff.py:11]: Are you sure it's ALL arm that they don't support? I thought it was just mac + arm

Comment (pauladkisson) [reply tests/test_scan_image_tiff.py:11]: I don't have a windows-arm machine handy, so I have no idea if it's mac + arm or just arm.  But for the sake of CI, does it really matter?

File: .github/workflows/run-tests.yml (hunk 2 above)
Comment (pauladkisson) [reply .github/workflows/run-tests.yml:83]: The cli-level ignore solution is more dry.  I am not particularly interested in adding pytest.mark.skipif to every test in tests/test_scanimagetiffimagingextractor.py (and then removing them if/when scanimage adds m1 mac support).

Comment (CodyCBakerPhD) [reply .github/workflows/run-tests.yml:83]: It's just as many lines to type and remove, and is easier to catch from source code in tests than to notice the split in CI

File: tests/test_scan_image_tiff.py (hunk 3 above)
Comment (CodyCBakerPhD) [reply tests/test_scan_image_tiff.py:11]: Well, if you wanted to specify JUST "CI" then you'd also fetch the CI environment variable from GitHub actions. The logic you have here affects every platform and device that might want to run the test suite, whereas we want a pretty narrow exclusion here

**Comment by pauladkisson** [comment]
wait, I'm in the middle of this fix too

**Comment by CodyCBakerPhD** [comment]
> @pauladkisson
[lowercase darwin](https://github.com/catalystneuro/roiextractors/pull/317/commits/a7c63278cf7688110724e352e528e8d4620fa38a)

I believe `platform.system()` is upper-case, `sys.platform` is lower case

You'll just also need a conditional add for https://github.com/catalystneuro/roiextractors/blob/workflow/tests/test_io.py#L48 to the list of IO to test

**Comment by pauladkisson** [comment]
> You'll just also need a conditional add for https://github.com/catalystneuro/roiextractors/blob/workflow/tests/test_io.py#L48 to the list of IO to test

Ah, right, forgot about that one.

**Comment by CodyCBakerPhD** [comment]
Looks great, thanks!

**Review by CodyCBakerPhD — APPROVED** [review]

**Pull request merged**